### PR TITLE
Implements the return of method optimize

### PR DESCRIPTION
**Pull request by samuelhgf**
Hi,

This PR implements a return for the method optimize. It will return an array that contains the binary name of each optimizer used with the status of the process. 
True if every occurred well or false if an error occurred.

This also implements an Exception that is throw when an optimizer used in the process is not installed.

### CODE REVIEW

**Comment by samuelhgf** [comment]
@freekmurze Could you see that PR, please?

File: src/OptimizerChain.php [hunk 1]
```diff
@@ -100,17 +106,32 @@ protected function applyOptimizer(Optimizer $optimizer, Image $image)
...
-    protected function logResult(Process $process)
+    protected function logResult(Process $process, Optimizer $optimizer)
     {
         if (! $process->isSuccessful()) {
+            $optimized = false;
+
             $this->logger->error("Process errored with `{$process->getErrorOutput()}`");
 
-            return;
+            if (strpos($process->getErrorOutput(), $optimizer->binaryName().': '.strtolower(Process::$exitCodes[127])) !== false) {
```
Comment (freekmurze) [inline src/OptimizerChain.php:119]: Doesn't the `canHandle` method avoid this situation?

File: src/OptimizerChain.php [hunk 2]
```diff
@@ -100,17 +106,32 @@ protected function applyOptimizer(Optimizer $optimizer, Image $image)
...
+                throw new OptimizerNotInstalledException('Optimized not installed!');
+
+                return;
+            }
+        } else {
```
Comment (freekmurze) [inline src/OptimizerChain.php:124]: Do we really need `else` here?

File: src/Optimizers/OptimizerNotInstalled.php [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace Spatie\ImageOptimizer\Optimizers;
+
+use Spatie\ImageOptimizer\Image;
+
+class OptimizerNotInstalled extends BaseOptimizer
+{
+    public $binaryName = 'not_installed';
+
+    /**
```
Comment (freekmurze) [inline src/Optimizers/OptimizerNotInstalled.php:11]: Remove this docblock.

File: src/Optimizers/OptimizerNotInstalled.php [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Spatie\ImageOptimizer\Optimizers;
```
Comment (freekmurze) [inline src/Optimizers/OptimizerNotInstalled.php:3]: Isn't this only used in the tests? If so, move it to the testsuite.

**Pull request closed without merging**